src/lib.rs

```diff
@@ -1,4 +1,170 @@
-//! A pure-rust TDS implementation for Microsoft SQL Server (>=2008)
+//! An asynchronous, runtime-independent, pure-rust TDS implementation for
+//! Microsoft SQL Server.
+//!
+//! # Connecting with async-std
+//!
+//! Being not bound to any single runtime, a `TcpStream` must be created
+//! separately and injected to the [`Client`].
+//!
+//! ```no_run
+//! use tiberius::{Client, ClientBuilder, AuthMethod};
+//! use async_std::net::TcpStream;
+//!
+//! #[async_std::main]
+//! async fn main() -> anyhow::Result<()> {
+//!     // Using the builder method to construct the options.
+//!     let mut config = ClientBuilder::new();
+//!
+//!     config.host("localhost");
+//!     config.port(1433);
+//!
+//!     // Using SQL Server authentication.
+//!     config.authentication(AuthMethod::sql_server("SA", "<password>"));
+//!
+//!     // Taking the address from the configuration, using async-std's
+//!     // TcpStream to connect to the server.
+//!     let tcp = TcpStream::connect(config.get_addr()).await?;
+//!     
+//!     // We'll disable the Nagle algorithm. Buffering is handled
+//!     // internally with a `Sink`.
+//!     tcp.set_nodelay(true)?;
+//!
+//!     // Handling TLS, login and other details related to the SQL Server.
+//!     let mut client = Client::connect(config, tcp).await?;
+//!
+//!     // A response to a query is a stream of data, that must be
+//!     // polled empty before querying again. Using streams allows
```

Comment:
Nitpick: I had to read twice to understand, maybe "polled until empty", or "polled to the end"?